google/cloud/pubsublite/internal/resumable_async_streaming_read_write_rpc_test.cc

```diff
@@ -214,6 +214,60 @@ TEST_F(ResumableAsyncReadWriteStreamingRpcTest,
   shutdown.get();
 }
 
+TEST_F(ResumableAsyncReadWriteStreamingRpcTest,
+       SingleStartFailurePermanentErrorWithAsyncLoop) {
+  struct Loop {
+    // this should run if and only if the stream is not shutdown
+    void InvokeLoop() {
+      stream->Read().then([this](future<absl::optional<FakeResponse>>) {
+        to_call();
+        if (status_future.is_ready() && !status_future.get().ok()) return;
```

Comment:
remove this.

Reply:
`!status_future.get().ok()`, you should stop looping even if it is OK (but fail the test)